### PR TITLE
Move data protection check to javascript

### LINKED ISSUE #20199: Admin panel incorrectly warns about data/.htaccess not working
When I open the admin panel, it tells me that my data directory would be available in public and the `.htaccess` file would not be working. However, when I actually try to access anything in the data directory (e.g., `/data/htaccesstest.txt`), then apache replies "Forbidden". So it seems that this is a false positive, but it still has me quite nervous. Can I be sure that the test is wrong here?

This is with ownCloud 8.2.0-2.1 on Debian Jessie.

### PR DESCRIPTION
**Pull request by vincchan**
fixes #20199

### CODE REVIEW

File: core/js/setupchecks.js [hunk 1]
```diff
@@ -194,6 +187,31 @@
 			return deferred.promise();
 		},
 
+		checkDataProtected: function() {
+			var deferred = $.Deferred();
+			if(oc_dataURL === false){
+				return deferred.resolve([]);
+			}
+			var afterCall = function(xhr) {
+				// console.log(xhr);
```
Comment (LukasReschke) [inline core/js/setupchecks.js]: Debug code. Can be removed :)

Comment (vincchan) [inline core/js/setupchecks.js]: Debug code removed :)

File: core/js/config.php [hunk 2]
```diff
@@ -62,9 +62,17 @@
 }
 $outgoingServer2serverShareEnabled = $config->getAppValue('files_sharing', 'outgoing_server2server_share_enabled', 'yes') === 'yes';
 
+$CountOfDataLocation = 0;
```
Comment (PVince81) [inline core/js/config.php]: please make the first letter of such variables lower case (it's by convention for PHP variables)

**Comment by PVince81** [comment]
Code looks good otherwise

**Comment by PVince81** [comment]
Raised https://github.com/owncloud/core/pull/22066 to wake up CI tests.

@vincchan Thanks a lot for your contribution!
Contributions to the core repo require a signed contributors agreement http://owncloud.org/about/contributor-agreement/

Alternatively you can add a comment here where you state that this contribution is MIT licensed.

Some more details about out pull request workflow can be found here: http://owncloud.org/code-reviews-on-github/

**Comment by vincchan** [comment]
@PVince81 CLA has been sent to frank@owncloud.com thanks!

**Comment by LukasReschke** [comment]
@karlitschek Please confirm. THX.

**Comment by karlitschek** [comment]
confirmed! Thanks a lot!

**Comment by LukasReschke** [comment]
CI passing at #22066

:+1: for this changeset from me. Thanks Vincent :smile: :rocket:

**Comment by PVince81** [comment]
:+1:

**Comment by RealRancor** [comment]
Really :-1: for this. Checking if the .ocdata is not accessible is no proof if the datadir is protected. Most webserver setups out there are denying the access to those "dot" files by default. There is no guarantee that the datadir is protected by the assumption that the .ocdata is accessible.

**Comment by RealRancor** [comment]
Raised https://github.com/owncloud/core/issues/24987

**Pull request merged**